nilearn/mass_univariate/tests/test_permuted_least_squares.py

```diff
@@ -356,126 +400,176 @@ def test_permuted_ols_withcovar_multivariate(random_state=0):
+
     # design parameters
     n_samples = 50
+    n_descriptors = 10
+    n_regressors = 1
+
     # create design
-    target_var = rng.randn(n_samples, 1)
-    tested_var = np.ones((n_samples, 1))
+    tested_var = np.ones((n_samples, n_regressors))
+    target_var = rng.randn(n_samples, n_descriptors)
+
     # compute t-scores with linalg or statmodels
     t_val_ref = get_tvalue_with_alternative_library(tested_var, target_var)
+    assert t_val_ref.shape == (n_regressors, n_descriptors)
+
     # permuted OLS
     neg_log_pvals, orig_scores, _ = permuted_ols(
         tested_var, target_var, confounding_vars=None, n_perm=10,
         random_state=random_state)
+    assert neg_log_pvals.shape == (n_regressors, n_descriptors)
+    assert orig_scores.shape == (n_regressors, n_descriptors)
     assert_array_less(neg_log_pvals, 1.)  # ensure sign swap is correctly done
+    assert_array_almost_equal(t_val_ref, orig_scores, decimal=6)
+
     # same thing but with model_intercept=True to check it has no effect
     _, orig_scores_addintercept, _ = permuted_ols(
-        tested_var, target_var, confounding_vars=None, model_intercept=True,
+        tested_var, target_var, confounding_vars=None, model_intercept=False,
         n_perm=0, random_state=random_state)
-    assert_array_almost_equal(t_val_ref, orig_scores, decimal=6)
-    assert_array_almost_equal(orig_scores, orig_scores_addintercept, decimal=6)
+    assert orig_scores_addintercept.shape == (n_regressors, n_descriptors)
+    assert_array_almost_equal(t_val_ref, orig_scores_addintercept, decimal=6)
 
 
 def test_permuted_ols_intercept_statsmodels_withcovar(random_state=0):
     rng = check_random_state(random_state)
+
     # design parameters
     n_samples = 50
+    n_descriptors = 10
+    n_regressors = 1
+    n_covars = 2
+
     # create design
-    target_var = rng.randn(n_samples, 1)
-    tested_var = np.ones((n_samples, 1))
-    confounding_vars = rng.randn(n_samples, 2)
+    tested_var = np.ones((n_samples, n_regressors))
+    target_var = rng.randn(n_samples, n_descriptors)
+    confounding_vars = rng.randn(n_samples, n_covars)
+
     # compute t-scores with linalg or statmodels
     ref_scores = get_tvalue_with_alternative_library(tested_var, target_var,
                                                      confounding_vars)
+    assert ref_scores.shape == (n_regressors, n_descriptors)
+
     # permuted OLS
     _, own_scores, _ = permuted_ols(
         tested_var, target_var, confounding_vars, n_perm=0,
         random_state=random_state)
+    assert own_scores.shape == (n_regressors, n_descriptors)
+    assert_array_almost_equal(ref_scores, own_scores, decimal=6)
+
     # same thing but with model_intercept=True to check it has no effect
     _, own_scores_intercept, _ = permuted_ols(
         tested_var, target_var, confounding_vars, model_intercept=True,
         n_perm=0, random_state=random_state)
-    assert_array_almost_equal(ref_scores, own_scores, decimal=6)
```

Comment:
why do remove that line ?

Reply:
I just moved it up to L501, so it would be located with the calculation of `own_scores`.